uaclient/livepatch.py

```diff
@@ -203,14 +213,20 @@ def is_kernel_supported(
         return bool(result.get("Supported", False))
 
 
-def _on_supported_kernel_cli() -> Optional[bool]:
+def _on_supported_kernel_cli() -> Optional[LivepatchSupport]:
     lp_status = status()
     if lp_status is None:
         return None
     if lp_status.supported == "supported":
-        return True
+        return LivepatchSupport.SUPPORTED
+    if lp_status.supported == "kernel-upgrade-required":
+        return LivepatchSupport.KERNEL_UPGRADE_REQUIRED
+    if lp_status.supported == "kernel-end-of-life":
+        return LivepatchSupport.KERNEL_EOL
     if lp_status.supported == "unsupported":
-        return False
+        return LivepatchSupport.UNSUPPORTED
+    if lp_status.supported == "unknown":
+        return LivepatchSupport.UNKNOWN
```

Comment:
@kian99 Can you confirm that these are the expected string values of the kernel support status that we could get from `canonical-livepatch status --format=json`?

Reply:
Ah gotcha, sounds good.